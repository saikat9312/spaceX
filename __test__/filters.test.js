import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import App from "../component/filters/index";

describe("Filter title", () => {
    it('App shows "Filters"', () => {
        const app = shallow(<App />);

        expect(app.find("h4").text()).toEqual("Filters");
    });
});

describe("With Snapshot Testing", () => {
    it('App shows "Filters"', () => {
        const component = renderer.create(<App />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Test Button component', () => {
    it('Test click event', () => {
        const mockCallBack = jest.fn();

        const button = shallow((<button onClick={mockCallBack}>Ok!</button>));
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });
});