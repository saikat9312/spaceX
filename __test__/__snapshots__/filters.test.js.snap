// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Filters" 1`] = `
<div
  className="filterContainer"
>
  <h4>
    Filters
  </h4>
  <div
    className="filterItems"
  >
    Launch Year 
    <hr />
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2006
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2007
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2008
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2009
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2010
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2011
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2012
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2013
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2014
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2015
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2016
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2017
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2018
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2019
    </button>
    <button
      className="filterButton"
      onClick={[Function]}
    >
      2020
    </button>
  </div>
  <br />
  <div
    className="filterItems"
  >
    Successful Launch 
    <hr />
    <div
      className="filterItems"
    >
      <button
        className="filterButton"
        onClick={[Function]}
      >
        True
      </button>
      <button
        className="filterButton"
        onClick={[Function]}
      >
        False
      </button>
    </div>
  </div>
  <br />
  <div
    className="filterItems"
  >
    Successful Landing 
    <hr />
    <div
      className="filterItems"
    >
      <button
        className="filterButton"
        onClick={[Function]}
      >
        True
      </button>
      <button
        className="filterButton"
        onClick={[Function]}
      >
        False
      </button>
    </div>
  </div>
</div>
`;
